### PR TITLE
Nkpatel/maxpool yolo v4 support

### PR DESCRIPTION
**Pull request by nilaykp**
### Ticket
https://github.com/tenstorrent/tt-metal/issues/7746

### Problem description
Add support for maxpool for highter filter/channel sizes. 

### What's changed
Support is added for filter lengths > 3 and channels > 256 for supporting YoLoV4

### Checklist
- [x] Post commit CI passes [link](https://github.com/tenstorrent/tt-metal/actions/runs/10791755830)
- [x] Device perf regressions and output report [LINK](https://github.com/tenstorrent/tt-metal/actions/runs/10791757920) 
- [] Model perf regressions and output report [LINK](https://github.com/tenstorrent/tt-metal/actions/runs/10791761965/job/29930450448)
- [x] T3000 frequent tests [LINK](https://github.com/tenstorrent/tt-metal/actions/runs/10791766097)
- [] T3000 demo tests [LINK](https://github.com/tenstorrent/tt-metal/actions/runs/10791782441)

### CODE REVIEW

File: tests/ttnn/unit_tests/operations/test_max_pool2d.py [hunk 1]
```diff
@@ -289,3 +289,149 @@ def test_pool_core_nondivis(
...
+        kernel_size=(kernel_h, kernel_w),
+        stride=(stride_h, stride_w),
+        padding=(pad_h, pad_w),
+        dilation=(dilation_h, dilation_w),
+        dtype=dtype,
+        device=device,
+        batch_size=in_n,
+        input_height=in_h,
+        input_width=in_w,
+        reader_patterns_cache=reader_patterns_cache,
```
Comment (mywoodstock) [inline tests/ttnn/unit_tests/operations/test_max_pool2d.py:389]: Use the new version `ttnn.max_pool2d_new`

File: ttnn/cpp/ttnn/operations/pool/maxpool/device/max_pool2d_multi_core_program_factory.cpp [hunk 2]
```diff
@@ -137,11 +137,14 @@ MaxPoolNew::MultiCore::cached_program_t max_pool_2d_multi_core_sharded_with_halo
             .set_globally_allocated_address(*reader_indices_buffer);
     auto in_reader_indices_cb = tt::tt_metal::CreateCircularBuffer(program, all_cores, in_reader_indices_cb_config);
 
+    auto in_cb_sz = (input_shape[3] * kernel_size_hw_padded) > (conv_op_utils::constants::TILE_HW * 8)
+                        ? (conv_op_utils::constants::TILE_HW * 8)
+                        : input_shape[3] * kernel_size_hw_padded;
```
Comment (mywoodstock) [inline ttnn/cpp/ttnn/operations/pool/maxpool/device/max_pool2d_multi_core_program_factory.cpp:142]: what is `8` here?

File: ttnn/cpp/ttnn/operations/pool/maxpool/device/max_pool2d_multi_core_program_factory.cpp [hunk 3]
```diff
@@ -168,6 +171,21 @@ MaxPoolNew::MultiCore::cached_program_t max_pool_2d_multi_core_sharded_with_halo
     auto in_tiled_cb = tt::tt_metal::CreateCircularBuffer(program, all_cores, in_tiled_cb_config);
     log_debug(tt::LogOp, "CB {} :: PS = {}, NP = {}", in_tiled_cb_id, in_tiled_cb_pagesize, in_tiled_cb_npages);
 
+    uint32_t max_pool_partials_cb_id = tt::CB::c_intermed1;  // max_pool partials
```
Comment (mywoodstock) [inline ttnn/cpp/ttnn/operations/pool/maxpool/device/max_pool2d_multi_core_program_factory.cpp:174]: can this CB creation made conditional for large window sizes only? that will keep the current mem usage same for existing models.

File: ttnn/cpp/ttnn/operations/pool/maxpool/device/max_pool2d_multi_core_program_factory.cpp (hunk 2 above)
Comment (nkpatel-tt) [reply ttnn/cpp/ttnn/operations/pool/maxpool/device/max_pool2d_multi_core_program_factory.cpp]: Maximum number of tiles which can be reduced at a time. Updated with proper name and comment for better understaning.

File: ttnn/cpp/ttnn/operations/pool/maxpool/device/max_pool2d_multi_core_program_factory.cpp (hunk 3 above)
Comment (nkpatel-tt) [reply ttnn/cpp/ttnn/operations/pool/maxpool/device/max_pool2d_multi_core_program_factory.cpp]: Done.

File: tests/ttnn/unit_tests/operations/test_max_pool2d.py (hunk 1 above)
Comment (nkpatel-tt) [reply tests/ttnn/unit_tests/operations/test_max_pool2d.py]: Done.

**Review by mywoodstock — APPROVED** [review]

**Comment by mywoodstock** [comment]
Also please rebase and combine all commits into one. And make sure pipelines pass.

**Review by eyonland — APPROVED** [review]

**Pull request merged**